client/src/components/EditProfile/ProfilePhoto/UploadPhoto.tsx

```diff
@@ -34,11 +34,13 @@ const UploadPhoto: React.FC = (): JSX.Element => {
         }
       });
     }
-  };
+  }, [selectedImage, updateSnackBarMessage]);
 
   useEffect(() => {
-    uploadImage();
-  });
+    if (selectedImage) {
```

Comment:
Currently both `uploadImage` and the `useEffect` checks for the same condition, would be tidier to do either or.